pkgs/top-level/stage.nix

```diff
@@ -269,14 +274,15 @@ let
     # Prefer appendOverlays if used repeatedly.
     extend = f: self.appendOverlays [f];
 
-    # Fully static packages.
-    # Currently uses Musl on Linux (couldn’t get static glibc to work).
+    # Fully static packages without shared libraries support.
+    # Currently uses Musl on Linux.
     pkgsStatic = nixpkgsFun ({
       overlays = [ (self': super': {
         pkgsStatic = super';
       })] ++ overlays;
       crossSystem = {
         isStatic = true;
+        hasSharedLibraries = false;
```

Comment:
Does dlopen not work in pkgsStatic?  IIRC, we can build shared objects for aarch64, but not x86_64, or something like that.

Reply:
Hm, perhaps renaming `isStatic` to `hasStaticLibraries` (in addition to `hasSharedLibraries`) and adding a new `hasDynamicLinker` attribute would be better (with appropriate asserts, i.e. `hasStaticLibraries || hasSharedLibraries`, `hasSharedLibraries -> hasDynamicLinker`)?

Reply:
I don't think that would make sense, because every environment has static libraries, including the default ones.